src/Provider/Provider.Web/Orchestrators/CloseVacancyOrchestrator.cs

```diff
@@ -21,14 +22,23 @@ public CloseVacancyOrchestrator(IProviderVacancyClient client, IRecruitVacancyCl
             _vacancyClient = vacancyClient;
         }
 
-        public async Task<CloseViewModel> GetCloseViewModelAsync(VacancyRouteModel vrm)
+        public async Task<CloseViewModel> GetCloseViewModelAsync(VacancyRouteModel vrm, VacancyUser user)
         {
             var vacancy = await _vacancyClient.GetVacancyAsync(vrm.VacancyId.GetValueOrDefault());
 
             Utility.CheckAuthorisedAccess(vacancy, vrm.Ukprn);
 
             if (!vacancy.CanClose)
-                throw new InvalidStateException(string.Format(ErrorMessages.VacancyNotAvailableForClosing, vacancy.Title));
+            {
+                throw new InvalidStateException(
+                    string.Format(ErrorMessages.VacancyNotAvailableForClosing, vacancy.Title),
+                    user,
+                    new Dictionary<string, string> 
```

Comment:
Maybe we could have an overload where we pass in the `Vacancy` and the `RecruitException` creates the dictionary for us.